lib/keila/files/files.ex

```diff
@@ -0,0 +1,175 @@
+defmodule Keila.Files do
+  @moduledoc """
+  Module for handling images and other files uploaded by users.
+  """
+
+  @max_size 8_000_000
+
+  use Keila.Repo
+  alias Keila.Files.File
+  alias __MODULE__.MediaType
+
+  @spec store_file(Project.id(), Path.t(), term) :: term | {:error, term()}
+  def store_file(project_id, source_path, metadata \\ []) do
+    filename = get_filename(source_path, metadata)
+    raw_type = Keyword.get(metadata, :type)
+
+    with {:ok, size} <- get_and_validate_size(source_path),
+         {:ok, type} <- get_and_validate_type(source_path, filename, raw_type) do
+      uuid = Ecto.UUID.generate()
+      sha256 = Keila.Hasher.hash_file(source_path, :sha256)
```

Comment:
Maybe the type hash can be configured in `config.exs`, but I don't think it's really important

Reply:
The SHA256 hash is also stored in the database (though it’s not currently used in any way … but). In the future this might be useful for features like deduplication though - or for searching for known files in the storage.